common/Anonymizer.hpp

```diff
@@ -156,6 +160,9 @@ class Anonymizer
     /// The only instance of the Anonymizer per process.
     static std::unique_ptr<Anonymizer> _instance;
 
+    /// The prefix counter.
+    std::atomic<unsigned> _prefix;
```

Comment:
This minimizes the scope of the lock. That lock can have a significant performance impact, so making this an atomic is a good idea. Otherwise, we have to take the lock over a much larger code (and for much longer time), or take it multiple times, locking and unlocking, which again is highly costly and problematic.